### PR TITLE
Added feature change industry placement

### PR DESCRIPTION
**Pull request by sajidmalik**

### CODE REVIEW

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Sfa.Tl.ResultsAndCertification.Common.Helpers;
+using Sfa.Tl.ResultsAndCertification.Common.Services.Cache;
+using Sfa.Tl.ResultsAndCertification.Web.ViewModel.AdminDashboard;
+using System.Threading.Tasks;
+using Xunit;
+namespace Sfa.Tl.ResultsAndCertification.Web.UnitTests.Controllers.AdminDashboardControllerTests.ChangeIndustryPlacementGet
+{
+    public class When_Called_With_Invalid_Data: TestSetup
+    {
+        public int PathwayId { get; set; }
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs:13]: Declared in parent class. Same with Result

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public IActionResult Result { get; private set; }
+
+
+        public override void Given()
+        {
+            PathwayId = 0;
+            AdminDashboardLoader.GetAdminLearnerRecordAsync<AdminChangeIndustryPlacementViewModel>(PathwayId).Returns(Mockresult);
+        }
+
+        public async override Task When()
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs:25]: The method is declared in the parent class

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+using Sfa.Tl.ResultsAndCertification.Web.Content.AdminDashboard;
+using Sfa.Tl.ResultsAndCertification.Web.ViewModel.AdminDashboard;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Sfa.Tl.ResultsAndCertification.Web.UnitTests.Controllers.AdminDashboardControllerTests.ChangeIndustryPlacementGet
+{
+    public class When_Called_With_Valid_Data : TestSetup
+    {
+        public int PathwayId { get; set; }
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs:14]: Lines 14to 16 can be removed. Declared in parent class

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                TlevelName = "t-level-name",
+                AcademicYear = 2020,
+                DisplayAcademicYear = "2021 to 2022",
+                IndustryPlacementStatus = Common.Enum.IndustryPlacementStatus.Completed
+            };
+
+            AdminDashboardLoader.GetAdminLearnerRecordAsync<AdminChangeIndustryPlacementViewModel>(PathwayId).Returns(Mockresult);
+        }
+
+        public async override Task When()
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs:39]: Can be removed. Declared in parent class

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Valid_Data.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            _internalApiClient = Substitute.For<IResultsAndCertificationInternalApiClient>();
+
+            var mapperConfig = new MapperConfiguration(c => c.AddMaps(typeof(AdminDashboardMapper).Assembly));
+            var mapper = new AutoMapper.Mapper(mapperConfig);
+
+            Loader = new AdminDashboardLoader(_internalApiClient, mapper);
+        }
+
+
+        private Models.OverallResults.OverallResultDetail _expectedOverallResult;
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Valid_Data.cs:33]: Not used at all

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Withdrawn_Learner.cs [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            _internalApiClient = Substitute.For<IResultsAndCertificationInternalApiClient>();
+
+            var mapperConfig = new MapperConfiguration(c => c.AddMaps(typeof(AdminDashboardMapper).Assembly));
+            var mapper = new AutoMapper.Mapper(mapperConfig);
+
+            Loader = new AdminDashboardLoader(_internalApiClient, mapper);
+        }
+
+
+        private Models.OverallResults.OverallResultDetail _expectedOverallResult;
```
Comment (ibai-gonzalez) [inline src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Withdrawn_Learner.cs:33]: Not used

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 7]
```diff
@@ -233,8 +233,33 @@ public async Task<IActionResult> ReviewChangeStartYearAsync(ReviewChangeStartYea
...
+        [Route("admin/admin-submit-change-industry-placement", Name = RouteConstants.AdminSubmitChangeIndustryPlacement)]
+        public async Task<IActionResult> ChangeIndustryPlacementAsync(AdminChangeIndustryPlacementViewModel model)
+        {
+            var viewModel = await _loader.GetAdminLearnerRecordAsync<AdminChangeIndustryPlacementViewModel>(model.PathwayId);
+
+            if (!ModelState.IsValid)
+            {
+                return View(model);
+            }
             await Task.CompletedTask;
```
[the remark this answers is not in the document]
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs]: This controller will change for the next user story as it's not done yet.

File: src/Sfa.Tl.ResultsAndCertification.Web/Mapper/AdminDashboardMapper.cs [hunk 8]
```diff
@@ -83,6 +82,20 @@ public AdminDashboardMapper()
                 .ForMember(d => d.TlevelName, opts => opts.MapFrom(s => s.TlevelName))
                 .ForMember(d => d.AcademicYear, opts => opts.MapFrom(s => s.AcademicYear))
                 .ForMember(d => d.DisplayAcademicYear, opts => opts.MapFrom(s => s.DisplayAcademicYear));
+
+            CreateMap<AdminLearnerRecord, AdminChangeIndustryPlacementViewModel>()
+                .ForMember(d => d.PathwayId, opts => opts.MapFrom(s => s.PathwayId))
```
[the remark this answers is not in the document]
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Mapper/AdminDashboardMapper.cs]: Done

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 9]
```diff
@@ -233,8 +233,33 @@ public async Task<IActionResult> ReviewChangeStartYearAsync(ReviewChangeStartYea
...
+                return RedirectToRoute(RouteConstants.PageNotFound);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("admin/admin-submit-change-industry-placement", Name = RouteConstants.AdminSubmitChangeIndustryPlacement)]
+        public async Task<IActionResult> ChangeIndustryPlacementAsync(AdminChangeIndustryPlacementViewModel model)
+        {
+            var viewModel = await _loader.GetAdminLearnerRecordAsync<AdminChangeIndustryPlacementViewModel>(model.PathwayId);
```
[the remark this answers is not in the document]
Comment (sajidmalik) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs]: This controller will change for the next user story as it's not done yet.

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementPost/TestSetup.cs [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+using System;
+using System.Threading.Tasks;
+
+namespace Sfa.Tl.ResultsAndCertification.Web.UnitTests.Controllers.AdminDashboardControllerTests.ChangeIndustryPlacementPost
+{
+    public abstract class TestSetup : AdminDashboardControllerTestBase
+    {
+        protected int AoUkprn;
+        protected int ProfileId;
+        protected Guid UserId;
```
[the remark this answers is not in the document]
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementPost/TestSetup.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementPost/TestSetup.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        protected ICacheService CacheService;
+        protected ILogger<AdminDashboardController> Logger;
+        protected AdminDashboardController Controller;
+        protected IProviderLoader ProviderLoader;
+        protected IHttpContextAccessor HttpContextAccessor;
+        protected AdminChangeIndustryPlacementViewModel AdminChangeIndustryPlacementViewModel;
+
+        public IActionResult Result { get; private set; }
+
+        public override void Setup()
```
[the remark this answers is not in the document]
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementPost/TestSetup.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs (hunk 2 above)
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Invalid_Data.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs (hunk 3 above)
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs (hunk 4 above)
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Controllers/AdminDashboardControllerTests/ChangeIndustryPlacementGet/When_Called_With_Valid_Data.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Valid_Data.cs (hunk 5 above)
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Valid_Data.cs]: Done

File: src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Withdrawn_Learner.cs (hunk 6 above)
Comment (sajidmalik) [reply src/Tests/Sfa.Tl.ResultsAndCertification.Web.UnitTests/Loader/AdminDashboardLoaderTests/AdminChangeIndustryPlacement/When_Called_With_Withdrawn_Learner.cs]: Done

**Review by ibai-gonzalez — APPROVED** [review]

**Pull request merged**